src/Compilers/CSharp/Test/Semantic/Semantics/LambdaDiscardParametersTests.cs

```diff
@@ -0,0 +1,309 @@
+                //             _) => 6L;
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "_").WithArguments("lambda discard parameters").WithLocation(17, 13),
+                // (20,13): error CS8652: The feature 'lambda discard parameters' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //             _,
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "_").WithArguments("lambda discard parameters").WithLocation(20, 13)
+                );
+
+            var tree = comp.SyntaxTrees.Single();
+            var underscores = tree.GetRoot().DescendantNodes().OfType<ParameterSyntax>().Where(p => p.ToString() == "_").ToArray();
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+            VerifyDiscardParameterSymbol(underscores[0], "System.Int16", CodeAnalysis.NullableAnnotation.NotAnnotated, model);
+            VerifyDiscardParameterSymbol(underscores[1], "System.String", CodeAnalysis.NullableAnnotation.None, model);
+        }
+
+        private static void VerifyDiscardParameterSymbol(ParameterSyntax underscore, string expectedType, CodeAnalysis.NullableAnnotation expectedAnnotation, SemanticModel model)
+        {
+            Assert.Null(model.GetSymbolInfo(underscore).Symbol);
+            var symbol1 = model.GetDeclaredSymbol(underscore);
+            Assert.Equal(expectedType, symbol1.Type.ToTestDisplayString());
+            Assert.Equal("_", symbol1.Name);
+            Assert.True(symbol1.IsDiscard);
+            Assert.Equal(expectedType, symbol1.Type.ToTestDisplayString());
+            Assert.Equal(expectedAnnotation, symbol1.NullableAnnotation);
+        }
+
+        [Fact]
+        public void DiscardParameters()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public static void Main()
+    {
+        System.Func<short, short, long> f1 = (_, _) => { long _ = 3; return _; };
+        System.Console.Write(f1(0, 0));
+
+        System.Func<short, short, int, long> f2 = (_, _, a) => 4L + a;
+        System.Console.Write(f2(0, 0, 1));
+
+        System.Func<int, short, short, long> f3 = (a, _, _) => 5L + a;
+        System.Console.Write(f3(1, 0, 0));
+    }
+}", options: TestOptions.DebugExe);
+
+            comp.VerifyDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "356");
+        }
+
+        [Fact]
+        public void DiscardParameters_RefAndOut()
+        {
+            var comp = CreateCompilation(@"
+class C
+{
+    delegate int RefAndOut(ref int i, out int j);
+    static void M()
+    {
+        RefAndOut f1 = (ref int _, out int _) =>
+            {
+                return 2;
+            };
+    }
+}");
+
+            // Note: this is somewhat problematic because there is nothing the user can do to fix this. We could have an error for out discards
```

Comment:
>We could have an error for out discards [](start = 104, length = 39)

Do we have a note for LDM to consider this? It definitely seems to me like it should be an error to have an out discard.

Reply:
It is always going to be an error, the question is just which error (ie. diagnostic quality) ;-)
So I'm okay to bring to LDM, but I think it's mostly a compiler decision.

Reply:
Took a note for LDM question in test plan